api/notd/token_metadata_processor.py

```diff
@@ -117,6 +117,21 @@ def _resolve_data(dataString: str, registryAddress: str, tokenId: str) -> Dict:
 
     @staticmethod
     async def _get_token_metadata_from_data(registryAddress: str, tokenId: str, metadataUrl: str, tokenMetadataDict: Dict[str, Any]) -> RetrievedTokenMetadata:
+        if not isinstance(tokenMetadataDict, dict):
```

Comment:
this should be outside before this function is called - the premise of this function is that the `tokenMetadataDict` is in fact a dict